README.md

```diff
@@ -2,3 +2,13 @@
 This repository is used to document and define coding practices such as coding styles, writing commit messages and documentation.
 
 The code_conventions.md defines the general coding conventions, without going into language specifics. It is possible that certain languages have their own coding standards. If this is the case, an additional document is added, which provides addendums or changes to the general codestyle.
+
+Vocabulary
+---
+ Solution for hotend versus nozzle and {}_index versus {}_nr or {}_number name usage:
+
+ * Only use 'hotend'
+ * Only use 'nozzle' to refer to (attributes of) the Olsson block at the end of the hotend
+ * Perform renaming according to the Boy-scout Rule, only in the same(/smallest) scope as where changes are made.
+ * Only use '{}_index' indicating an index, counting value, starting at zero (0), referring to a position in an iterable. This will mostly be the case.
+ * Only use '{}_number' indicating a counting value, starting at one (1), referring to a ordinal number (e.g. first, second, third) nearly always used for communication to a user.
```

Comment:
I would not advise to put down rules for this. There are always exceptions. A counting value doesn't always start at 1 (you could have 0 walls for instance). And indices do not always start at 0 (like how raft layers have a negative layer index).

Reply:
There are and always will be exceptions, possibly for each and every rule in this file, and every other file in the meta repository. This leaves us with a few options: only write stuff down without any exception and do not review based on everything or accept possible long discussions/reviews that depend greatly on the person reviewing. Or write down each and every possible exception in such a way that there can be absolutely no doubt as to what the rules are. Or we write down the choices we make that cover the bulk of the work and have discussions about the exceptions.

Reply:
These should be defined in the generic code definitions, not in python.

Reply:
Misread, moved to code_conmventions.md